src/aff_expr.jl

```diff
@@ -292,7 +308,7 @@ function isequal_canonical(aff::GenericAffExpr{C,V}, other::GenericAffExpr{C,V})
 end
 
 Base.convert(::Type{GenericAffExpr{T,V}}, v::V)    where {T,V} = GenericAffExpr(zero(T), v => one(T))
-Base.convert(::Type{GenericAffExpr{T,V}}, v::Real) where {T,V} = GenericAffExpr{T,V}(convert(T, v))
+Base.convert(::Type{GenericAffExpr{T,V}}, v::_Constant) where {T,V} = GenericAffExpr{T,V}(convert(T, _constant_to_number(v)))
```

Comment:
This is a long line. Consider using function instead of one-liners for better readability.